### PR TITLE
#163902213 Created user login endpoint

### PR DESCRIPTION
**Pull request by carlos-nyaga**
#### What does this PR do?
Creates a fully functional /api/v2/auth/login endpoint

#### Description of Task to be completed?
A registered user is able to login by sending a POST request  with valid credentials to  the endpoint

#### How should this be manually tested?
Once cloned the repo, cd to the root directory and run `python manage.py` then using postman or preferred client send a POST request to:
> http://127.0.0.1:5000/api/v2/auth/login
> This returns a session token

#### Any background context you want to provide?
Currently working on v2 of the api

#### What are the relevant pivotal tracker stories?
[#163902213] (https://www.pivotaltracker.com/story/show/163902213)

screenshot_2019-02-15_00-04-30

### CODE REVIEW

File: app/api_2/views/users.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    oname = data['other_name']
+    email = data['email']
+    phoneNo = data['phone_number']
+    passport = data['passport_url']
+    password = data['password']
+    isadmin = data['isadmin']
+
+    created_user = Users().user_create(fname, lname, oname, email, phoneNo, passport,password, isadmin)
+
+    return make_response(jsonify({
```
Comment (Tittoh) [inline app/api_2/views/users.py:38]: Avoid too many `return` statements within this function. validation can be abstracted

File: app/api_2/views/users.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+@bp2.route('/auth/login', methods = ['POST'])
+def login():
+    data = request.get_json() or {}
+    if "password" not in data or "email" not in data:
+        return bad_request('must include name and email fields')
+    
+    if any(map(lambda x: len("".join(str(x).split())) < 1, [data["password"], data["email"]])):
+        return bad_request('Nice try...but no....we do not accept blanks')
+
+    elif any(map(lambda x: len("".join(str(x).split())) < 3, [data["password"], data["email"]])):
```
Comment (Tittoh) [inline app/api_2/views/users.py:53]: same here, the validation from signup can be reused

File: app/api_2/views/users.py (hunk 1 above)
Comment (carlos-nyaga) [reply app/api_2/views/users.py:38]: Alright....working on it

File: app/api_2/views/users.py (hunk 2 above)
Comment (carlos-nyaga) [reply app/api_2/views/users.py:53]: working on it

**Pull request merged**